src/config.c

```diff
@@ -1693,14 +1693,16 @@ config_check ( void )
 
 static int config_newcfg = 0;
 
-static char *config_get_dir ( void )
+static char *config_get_dir ( uid_t uid )
 {
-  char hts_home[PATH_MAX + sizeof("/.hts/tvheadend")]; /* Must be largest of the 3 config strings! */
```

Comment:
Why is this needed?

hts_home is either something (usually $HOME) + `/.hts/tvheadend` OR `/var/lib/tvheadend`, which fits into PATH_MAX just fine.

Reply:
Yeah, but we never do $HOME/var/lib/tvheadend :) the goal is, to have `hts_home` be large enough. Ideally, the var stays at `PATH_MAX`, and everything else gets reduced to fit, but that's far more complex :p this is much lazyer ..easier :)

So we need room to store $HOME + whatever; OR `/etc/tvheadend` | `/var/lib/tvheadend`. The later two, easily fit within `PATH_MAX`.

Reply:
Good point, I had failed to consider that.  I changed it back.